From d891c52c1ca8ffae0aa9cebea572dccba1186355 Mon Sep 17 00:00:00 2001
From: Bruno Haible <bruno@clisp.org>
Date: Tue, 20 Aug 2024 03:27:43 +0200
Subject: [PATCH] maint: Avoid a gcc 14 warning that makes
 --enable-gcc-warnings break.

* src/output.c: Disable -Wnull-dereference warnings in this file.
---
 src/output.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/output.c b/src/output.c
index 6dc8280a..65f183dd 100644
--- a/src/output.c
+++ b/src/output.c
@@ -27,6 +27,12 @@
 #include "gl_avltree_oset.h"
 #include "gl_xoset.h"
 
+/* Work around a bogus GCC warning
+   <https://gcc.gnu.org/bugzilla/show_bug.cgi?id=116426>.  */
+#if __GNUC__ >= 6
+# pragma GCC diagnostic ignored "-Wnull-dereference"
+#endif
+
 /* Size of initial in-memory buffer size for diversions.  Small diversions
    would usually fit in.  */
 #define INITIAL_BUFFER_SIZE 512
-- 
2.34.1

